feat(changelog): new post (#173)
deta/space-docs: src/pages/changelog/post-15.mdx

---
title: "Improved Data and Resource Management"
date: "16 Aug 2023"
layout: "@changelogs"
description: "You now have more control over your data and have the ability to delete any resource you create. You can also update environment variables from the CLI and have longer sessions."
---

import DetaHeart from '@/components/DetaHeart.astro'
import Image from '@/components/Image.astro'

> ## Longer Sessions

Reviewer: Can we put this section at bottom?

Author: thought this is something a lot of users cared about and since it is short I put it at the top

Reviewer: Also bundle it with `space env` so that all the delete stuff is together.

But we could also bring `space env` up here. I liked how you ended on delete account

Author: yeah I also had both at the end but liked the ending with delete account.

> A lot of you have been asking for longer sessions and we have listened. After logging in to Space, you will now be logged in for 30 days. This means you will be able to access your account and your apps without having to log in again for longer. You can still log out manually if you need to.

Reviewer: ```suggestion
One of the biggest complaints with Deta was short sessions, requiring you to constantly log in. We extended the sessions, so after logging in to Space, you will stay logged in for 30 days. You can still log out manually if you need to.
```

## Manage your Data

### Exporting Data

> Core to the idea of a personal cloud computer is the ability to have control over your data. This also includes being able to export your data from Space to use for what ever you want.

Reviewer: ```suggestion
Core to the idea of a personal cloud computer is the ability control your own data. This includes being able to take your data with you outside of Space, to use for whatever you want.
```

<Image src="/changelog_assets/export-data.png" alt="Export Data" width="90%" background />

> You can now export your data from the "Data" view of your app instance. This will export all Bases and Drives that your app uses to a ZIP file that you can download.

Reviewer: ```suggestion
You can now export an app's data from the "Data" view of any app instance. You can also export the data in any Collection from its Settings. An export will export all Bases and Drives to a ZIP file that you can download.
```

> ### Clear App Data

Reviewer: ```suggestion
### Wipe App Data
```

> Want to start fresh? You can now clear all the data in your app instance. This will delete all the Bases and Drives that your app uses. You can do this from the "Data" view of your app instance as well.

Reviewer: ```suggestion
Want to start fresh? You can now wipe all the data in your app instance from the "Data" view of your app instance. This will delete all the Bases and Drives that your app uses. 
```

<Image src="/changelog_assets/clear-app-data.png" alt="Clear App Data" width="90%" background />

### Delete App, Keep Data

> Sometimes, you want the opposite of wiping the data, which is deleting the app, but keeping the data. Sounds wild, right? Dump the app, but cherish the data in a collection.

Reviewer: ```suggestion
Sometimes, you want the opposite of wiping the data, which is deleting the app, but keeping the data. Sounds wild, right? From your app's "Settings" menu you can now export your data to a Collection whenever you delete an app.
```

<Image src="/changelog_assets/delete-app-keep-data.png" alt="Delete App" width="90%" background />

## Managing your Project

### Delete Releases

> Ever released something by accident or want to delete an old release? You can now do that from the "Publish" tab in your app's dashboard.

Reviewer: ```suggestion
Ever released something by accident or want to delete an old release? You can now do that from the "Publish" tab in your project's dashboard in Builder.
```

<Image src="/changelog_assets/manage-release.png" alt="Manage Release" width="90%" background />

> If the release doesn't have any installations yet, you can delete it. If it does, you can unlist it from Discovery instead. This will remove it from the app's page and the Discovery homepage. Existing installations will still be able to use it.

Reviewer: ```suggestion
If the release doesn't have any installations yet, you can delete it. If it has installations, you can unlist it from Discovery instead. This will remove the app's Discovery page and the app from the Discovery homepage. Existing users who installed the app will still be able to use their copy.
```

### Delete Project

> You are now able to delete your entire project. This will delete all the releases and installations of your app. If your app is listed on Discovery and has existing installations, you won't be able to delete it for now. We are working on a way to do this safely.

Reviewer: ```suggestion
You can now also delete an entire project in Builder. This will delete all the releases of your app. If your app is listed on Discovery and others have installed it, you won't be able to delete it for now. We are working on a way to do this safely.
```

<Image src="/changelog_assets/unlist-project.png" alt="Unlist Project" width="90%" background />

> ### Unlist Project from Discovery

Reviewer: ```suggestion
### Unlist a Project from Discovery
```

Reviewer: would also move right under delete releases

Author: had it there first but decided to put it here since it shares the same screenshot with delete project

> If you want to hide your entire app from Discovery instead, you can do that from the same menu. This will remove the app and all releases from being listed publicly on Discovery. Existing installations will continue working.

Reviewer: ```suggestion
In addition to individual releases, you can unlist an entire app from a project's Settings menu in Builder. This will remove the app and all releases from being listed publicly. Existing users who have installed your app will continue to be able to use it.
```

## Environment Variables and the CLI

> You can now use the `space builder env` command of the [Space CLI](https://deta.space/docs/en/build/reference/cli) to fetch or update the environment variables in your project's development instance.

Reviewer: ```suggestion
You can now use the `space builder env` command of the [Space CLI](https://deta.space/docs/en/build/reference/cli) to fetch or update the environment variables in your project's Builder instance.
```

Fetching them:

```bash

> space builder env --get .env

Reviewer: why do i have to pass `.env` in the getter function?

@deta/omer / @BetaHuhn

Reviewer: ah got it

```

Update them:

```bash

> space builder env --set .env

Reviewer: can i pass a custom arg for a diff file?

Reviewer 2: What exactly do you mean?

Reviewer: Ok I think I understand, so if my local env file is `.env.local` i can use `space builder env --set .env.local`

Reviewer 2: yep, definitely!

```

> Note that this is completely decoupled from your local [Spacefile](/docs/en/reference/spacefile/), and you need to push a new development instance if you add or remove a custom environment variables.

Reviewer: What does this mean, push a new dev instance if i add / remove custom env vars?

Author: Took this from [the docs for the command](https://github.com/deta/space-docs/pull/162/files). I think it refers to adding a new custom environment variable in which case you have to specify it in the Spacefile. Maybe @brkp can expand on it

Reviewer 2: so if you add a new custom env variable to your Spacefile, you need to push your application once before you can start using that variable with `space builder env`. this is the case because it's not trivial to add new env variables to the lambdas

## Deleting Resources

### Delete Collections

> In addition to all the other resources, you can now delete collections as well. This will delete all the data in the collection and remove it from your Space.

Reviewer: ```suggestion
In addition to all the other resources, you can now delete Collections as well. This will delete all the data in a Collection and remove it from your Space.
```

<Image src="/changelog_assets/delete-collection.png" alt="Unlist Project" width="90%" background />

> You can do this from the "Settings" menu of your collection.

Reviewer: ```suggestion
You can do this from the "Settings" menu of your Collection.
```
